src/lib/list/selection-list.ts

```diff
@@ -518,11 +518,17 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
     // emit the changed event when something actually changed.
     let hasChanged = false;
 
-    this.options.forEach(option => {
-      if (option._setSelected(isSelected)) {
-        hasChanged = true;
-      }
-    });
+    // Don't change selected state if MatSelectionList is disabled
+    if (!this.disabled) {
```

Comment:
This check should be removed now. Since we now check each individual option, this check just makes the `!option.disabled` condition a noop.

I'd format it like this:

```ts
    this.options.forEach(option => {
      if (!option.disabled && option._setSelected(isSelected)) {
        hasChanged = true;
      }
    });
```

Reply:
Also this shouldn't be done in this method, but rather inline inside the `keydown` handler. This won't allow for all options to be selected/deselected programmatically.